python/pyspark/mllib/linalg.py

```diff
@@ -1152,9 +1156,385 @@ def sparse(numRows, numCols, colPtrs, rowIndices, values):
+        return CoordinateMatrix(jcm)
+
+
+class RowMatrix(DistributedMatrix):
+    """
+    Represents a row-oriented distributed Matrix with no meaningful row indices.
+
+    .. note:: Experimental
+    """
+    def __init__(self, jrm):
+        """ Create a wrapper over a Java RowMatrix. """
+        self._jrm = jrm
+        self.rows = self._rows()
+
+    def _rows(self):
+        """
+        Get the rows of the RowMatrix as a RDD of Vectors.
+
+        >>> rows = sc.parallelize([Vectors.dense([1, 2, 3]), Vectors.dense([4, 5, 6]),
+        ...                        Vectors.dense([7, 8, 9]), Vectors.dense([10, 11, 12])])
+        >>> rm = DistributedMatrices.rowMatrix(rows)
+        >>> rowsRDD = rm.rows
+        >>> rowsRDD.first()
+        DenseVector([1.0, 2.0, 3.0])
+        """
+        return self._jrm.call("rows")
+
+    def numRows(self):
+        """
+        Get or compute the number of rows.
+
+        >>> rows = sc.parallelize([Vectors.dense([1, 2, 3]), Vectors.dense([4, 5, 6]),
+        ...                        Vectors.dense([7, 8, 9]), Vectors.dense([10, 11, 12])])
+        >>> rm = DistributedMatrices.rowMatrix(rows)
+        >>> int(rm.numRows())
+        4
+
+        >>> rows = sc.parallelize([Vectors.dense([1, 2, 3]), Vectors.dense([4, 5, 6]),
+        ...                        Vectors.dense([7, 8, 9]), Vectors.dense([10, 11, 12])])
+        >>> rm = DistributedMatrices.rowMatrix(rows, 7, 6)
+        >>> int(rm.numRows())
+        7
+        """
+        return self._jrm.call("numRows")
+
+    def numCols(self):
+        """
+        Get or compute the number of cols.
+
+        >>> rows = sc.parallelize([Vectors.dense([1, 2, 3]), Vectors.dense([4, 5, 6]),
+        ...                        Vectors.dense([7, 8, 9]), Vectors.dense([10, 11, 12])])
+        >>> rm = DistributedMatrices.rowMatrix(rows)
+        >>> int(rm.numCols())
+        3
+
+        >>> rows = sc.parallelize([Vectors.dense([1, 2, 3]), Vectors.dense([4, 5, 6]),
+        ...                        Vectors.dense([7, 8, 9]), Vectors.dense([10, 11, 12])])
+        >>> rm = DistributedMatrices.rowMatrix(rows, 7, 6)
+        >>> int(rm.numCols())
+        6
+        """
+        return self._jrm.call("numCols")
+
+
+class IndexedRow(object):
+    """
+    Represents a row of an IndexedRowMatrix.  Just a wrapper over a (long, Vector) tuple.
+
+    .. note:: Experimental
+    """
+    def __init__(self, index, vector):
+        self.index = long(index)
+        self.vector = _convert_to_vector(vector)
+
+    def __repr__(self):
+        return "IndexedRow(%s, %s)" % (self.index, self.vector)
+
+
+class IndexedRowMatrix(DistributedMatrix):
+    """
+    Represents a row-oriented distributed Matrix with indexed rows.
+
+    .. note:: Experimental
+    """
+    def __init__(self, jirm):
+        """ Create a wrapper over a Java IndexedRowMatrix. """
+        self._jirm = jirm
+        self.rows = self._rows()
+
+    def _rows(self):
```

Comment:
Yeah, the `_rows()` method is private since it is just the implementation for the variable `self.rows`, which mimics the `val rows` in Scala.  Pulling the logic out just makes the code cleaner, which is more important for `IndexedRowMatrix` and `CoordinateMatrix` than for `RowMatrix`.